docs/src/main/tut/task.md

```diff
@@ -0,0 +1,20 @@
+---
+menu: tut
+weight: 115
+title: Task[T]
+---
+
+As you might have noticed, a few methods, e.g. `withBody` lift a `Response` into a `Task` context, so you end up with a `Task[Response]`. Coincidentially, all the methods which relate to the body of the `Request`/`Response`. Why does this happen? http4s is built on [Streaming], so directly materializing the body of a `Response` defeats any streaming benefits. Instead, the result is wrapped in a `Task` which preserves the asynchronous nature.
```

Comment:
Maybe s/coincidentally/namely/.  It's not a coincidence.

I'd say it's less about streaming and more about effects: we might know the content length before we start streaming bytes, but not until we stat the file or hear back from the database.  Both the streaming and the delayed content length calculation are about delaying side effects.